cmake/projects/libsodium/schemes/url_sha1_libsodium_msbuild.cmake.in

```diff
@@ -0,0 +1,117 @@
+cmake_minimum_required(VERSION 3.0)
+project(Hunter)
+
+
+# Inludes
+list(APPEND CMAKE_MODULE_PATH "@HUNTER_SELF@/cmake/modules")
+include(ExternalProject)
+include(hunter_install_libsodium_config)
+include(hunter_user_error)
+include(hunter_status_debug)
+include(hunter_test_string_not_empty)
+
+
+# Check preconditions
+hunter_test_string_not_empty("@HUNTER_EP_NAME@")
+hunter_test_string_not_empty("@HUNTER_INSTALL_PREFIX@")
+hunter_test_string_not_empty("@HUNTER_libsodium_VERSION@")
+hunter_test_string_not_empty("@HUNTER_MSVC_ARCH@")
+hunter_test_string_not_empty("@HUNTER_MSVC_VCVARSALL@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_URL@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_SHA1@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_DOWNLOAD_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_SOURCE_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_INSTALL_PREFIX@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_CONFIGURATION_TYPES@")
+hunter_test_string_not_empty("@HUNTER_SELF@")
+
+
+if(NOT "@MSVC@")
+    hunter_user_error("MSBuild scheme only supported with Visual Studio")
+endif()
+
+# Check MSVC Platform Toolset
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "10" _is_platform_toolset_v100)
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "11" _is_platform_toolset_v110)
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "12" _is_platform_toolset_v120)
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "14" _is_platform_toolset_v140)
+
+if(_is_platform_toolset_v100)
+    set(msvc_platform_toolset "v100")
+elseif(_is_platform_toolset_v110)
+    set(msvc_platform_toolset "v110")
+elseif(_is_platform_toolset_v120)
+    set(msvc_platform_toolset "v120")
+elseif(_is_platform_toolset_v140)
+    set(msvc_platform_toolset "v140")
+else()
+    hunter_user_error("Visual studio version not supported. Supported versions are: Visual studio 2010, 2011, 2012 and 2014.")
+    hunter_user_error("If there is a new version please open an issue at https://github.com/ruslo/hunter and mention @Cyberunner23")
```

Comment:
Second `hunter_user_error` message will not be printed since `hunter_user_error` will stop the flow similar to `FATAL_ERROR`. You can split long line like that:
```cmake
hunter_user_error(
    "A"
    "B"
    "C"
)
```